tests/v2/validation/nodescaling/replace.go

```diff
@@ -24,22 +24,24 @@ const (
 	clusterLabel                 = "cluster.x-k8s.io/cluster-name"
 )
 
-func MatchNodeToRole(t *testing.T, client *rancher.Client, clusterID string, isEtcd bool, isControlPlane bool, isWorker bool) (int, *management.Node) {
+func MatchNodeToRole(t *testing.T, client *rancher.Client, clusterID string, isEtcd bool, isControlPlane bool, isWorker bool) (int, []management.Node) {
 	machines, err := client.Management.Node.List(&types.ListOpts{Filters: map[string]interface{}{
 		"clusterId": clusterID,
 	}})
 	require.NoError(t, err)
+
 	numOfNodes := 0
-	lastMatchingNode := &management.Node{}
+	matchingNodes := []management.Node{}
 
 	for _, machine := range machines.Data {
 		if machine.Etcd == isEtcd && machine.ControlPlane == isControlPlane && machine.Worker == isWorker {
-			lastMatchingNode = &machine
+			matchingNodes = append(matchingNodes, machine)
 			numOfNodes++
 		}
 	}
-	require.NotEmpty(t, lastMatchingNode.NodeName, "matching node name is empty")
-	return numOfNodes, lastMatchingNode
+	require.NotEmpty(t, matchingNodes, "matching node name is empty")
+
+	return numOfNodes, matchingNodes
```

Comment:
nit; returning numOfNodes is redundant now since you can get this from the returned list instead.

Reply:
The `IsNodeReplaced` function is specifically calling that though, so it needs it, right?

Reply:
we could leave it if refactoring it is too much time, but usually if you return a list you would assume that the function(s) receiving the list can get its length if needed

Reply:
Sounds good, that can be future enhancement I think, that might need a bit more design discussion that is out of the scope.